lib/query/arith.js

```diff
@@ -0,0 +1,258 @@
+/*-
```

Comment:
I assume this class and related changes are delayed implementation of the arithmetic operators already in other SDKs, correct?

Reply:
No, these were already there, I just moved them to the new file to avoid circular dependency.